src/DependencyInjection/Compiler/RegisterMappingDriversPass.php

```diff
@@ -18,7 +18,9 @@ public function process(ContainerBuilder $container): void
             new Reference('vich_uploader.metadata_driver.xml'),
         ];
 
-        if ($container->has('annotation_reader')) {
+        $metaDataType = $container->getParameter('vich_uploader.metadata_type');
+
+        if ('attribute' === $metaDataType || ('annotation' === $metaDataType && $container->has('annotation_reader'))) {
```

Comment:
I don't see a case in which this condition could be false: the metadata type defaults to "attribute" and it can be only "attribute" or "annotation", the container has always "annotation_reader"

Reply:
@garak I thought it could be `xml` or `yaml` as well?

Reply:
OK, sorry, you're right, the presence of "annotation_reader" is not ensured.
But so at this point, I don't get why you need to load the managers into the `vich_uploader.metadata_driver.annotation` service. If they're only needed when using annotations and you're using attributes, you should be OK without them

Reply:
I'm really not sure - I was trying to make a minimal change to fix the issue at hand and keep the status quo, rather than making more wholesale changes.